sorting/random_pivot_quick_sort.cpp

```diff
@@ -0,0 +1,307 @@
+/**
+ * @file
+ * @brief A cpp program for random pivot quick sort algorithm. 
+ * [Random Pivot Quick Sort](https://www.sanfoundry.com/cpp-program-implement-quick-sort-using-randomisation)
+ * @details Intro :
+ *              - A random pivot quick sort algorithm is pretty much same as quick sort with a difference of having a logic of
+ *                  selecting next pivot element from the input array.
+ *              - Where in quick sort is fast, but still can give you the time complexity of O(n^2) in worst case.
+ *              - To avoid hitting the time complexity of O(n^2), we use the logic of randomize the selection process of pivot
```

Comment:
Use `*` instead of `-` (same in other parts of the code).
```suggestion
 *              * A random pivot quick sort algorithm is pretty much the same as quick sort with a difference of having a logic of
 *                  selecting the next pivot element from the input array.
 *              * Where in quick sort is fast, but still can give you the time complexity of O(n^2) in the worst case.
 *              * To avoid hitting the time complexity of O(n^2), we use the logic of randomizing the selection process of the pivot
```